### PR TITLE
Set Docker permissions during the build process

### LINKED ISSUE #3194: Permission update on docker entrypoint takes a long time
With the current approach on docker entrypoint for updating the files to the new custom UID/GID it takes forever to finish the process, which timeouts in a reasonable production container health check.

Why not just use `chown -rf mastodon:mastodon /mastodon/public/system` instead of finding and filtering non-mastodon files?

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

### PR DESCRIPTION
**Pull request by Gargron**
Fix #3194

I'm not sure why @Wonderfall didn't open this PR...

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -70,10 +72,11 @@ RUN bundle config build.nokogiri --with-iconv-lib=/usr/local/lib --with-iconv-in
 
 COPY . /mastodon
```
Comment (moritzheiber) [inline Dockerfile]: You can now [run chown while copying](https://docs.docker.com/engine/reference/builder/#copy), which is more efficient and saves you an additional run statement. I'd get rid of the `chown` statement in line 76 in favor of using `COPY --chown=${UID}:${GID}`?

**Comment by moritzheiber** [comment]
Are we still using `docker-entrypoint.sh` after this PR? Can we get rid of it?

**Review by moritzheiber — COMMENTED** [review]
Are we still using `docker-entrypoint.sh` after this PR? Can we get rid of it?

File: Dockerfile (hunk 1 above)
Comment (Wonderfall) [reply Dockerfile]: We could give it a try, indeed.

**Pull request closed without merging**